Mage.Sets/src/mage/cards/r/RottenmouthViper.java

```diff
@@ -0,0 +1,164 @@
+        Ability ability = new EntersBattlefieldOrAttacksSourceTriggeredAbility(
+                new AddCountersSourceEffect(CounterType.BLIGHT.createInstance())
+        );
+        ability.addEffect(new RottenmouthViperEffect(new CountersSourceCount(CounterType.BLIGHT)));
+        this.addAbility(ability);
+    }
+
+    private RottenmouthViper(final RottenmouthViper card) {
+        super(card);
+    }
+
+    @Override
+    public RottenmouthViper copy() {
+        return new RottenmouthViper(this);
+    }
+}
+
+class RottenmouthViperEffect extends OneShotEffect{
+
+    DynamicValue blightCounterAmount;
+
+    RottenmouthViperEffect(DynamicValue blightCounterAmount) {
+        super(Outcome.LoseLife);
+        this.blightCounterAmount = blightCounterAmount;
+        this.staticText = "Then for each blight counter on it, each opponent loses 4 life unless that player sacrifices a nonland permanent or discards a card.";
+    }
+
+    private RottenmouthViperEffect(final RottenmouthViperEffect effect) {
+        super(effect);
+        this.blightCounterAmount = effect.blightCounterAmount.copy();
+    }
+
+    @Override
+    public RottenmouthViperEffect copy() {
+        return new RottenmouthViperEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller != null){
+            int repeat = blightCounterAmount.calculate(game, source, this);
+            for (int i = 1; i <= repeat; i++) {
+
+                for (UUID opponentId : game.getOpponents(source.getControllerId())) {
+                    Player opponent = game.getPlayer(opponentId);
+                    if (opponent != null) {
+                        int permanents = game.getBattlefield().countAll(StaticFilters.FILTER_PERMANENT_NON_LAND, opponentId, game);
+                        if (permanents > 0 && opponent.chooseUse(outcome, "Sacrifices a nonland permanent? (Iteration " + i + " of " + repeat + ")",
+                                "Otherwise you have to discard a card or lose 4 life.", "Sacrifice", "Discard or life loss", source, game)) {
+                            Target target = new TargetPermanent(StaticFilters.FILTER_CONTROLLED_PERMANENT_NON_LAND);
+                            target.withNotTarget(true);
+                            if (opponent.choose(outcome, target, source, game)) {
+                                Permanent permanent = game.getPermanent(target.getFirstTarget());
+                                if (permanent != null) {
+                                    if (permanent.sacrifice(source, game)) {
+                                        continue;
+                                    }
+                                }
+                            }
+                        }
+                        if (!opponent.getHand().isEmpty() && opponent.chooseUse(outcome, "Discard a card? (Iteration " + i + " of " + repeat + ")",
```

Comment:
Same here. Must Outcome.Benefit so AI will discard instead loose life all the time.